.buildkite/x-pack/pipeline.xpack.heartbeat.yml

```diff
@@ -25,39 +25,189 @@ env:
+  #   steps:
+  #     - label: ":windows: Windows 10 Unit Tests"
+  #       command: |
+  #         Set-Location -Path $BEATS_PROJECT_NAME
+  #         mage build unitTest
+  #       key: "extended-win-10-unit-tests"
+  #       agents:
+  #         provider: "gcp"
+  #         image: "${IMAGE_WIN_10}"
+  #         machineType: "${GCP_WIN_MACHINE_TYPE}"
+  #         disk_size: 100
+  #         disk_type: "pd-ssd"
+  #       artifact_paths:
+  #         - "$BEATS_PROJECT_NAME/build/*.xml"
+  #         - "$BEATS_PROJECT_NAME/build/*.json"
+  #       notify:
+  #         - github_commit_status:
+  #             context: "$BEATS_PROJECT_NAME: Windows 10 Unit Tests"
 
-  - input: "Input Parameters"
-    key: "force-run-stages"
-    fields:
-    - select: "Heartbeat - run_xpack_heartbeat"
-      key: "run_xpack_heartbeat"
-      options:
-        - label: "True"
-          value: "true"
-        - label: "False"
-          value: "false"
-      default: "false"
-    - select: "Heartbeat - run_xpack_heartbeat_macos_tests"
-      key: "run_xpack_heartbeat_macos_tests"
-      options:
-        - label: "True"
-          value: "true"
-        - label: "False"
-          value: "false"
-      default: "false"
-
-    if: "build.source == 'ui'"
+  #     - label: ":windows: Windows 11 Unit Tests"
+  #       command: |
+  #         Set-Location -Path $BEATS_PROJECT_NAME
+  #         mage build unitTest
+  #       key: "extended-win-11-unit-tests"
+  #       agents:
+  #         provider: "gcp"
+  #         image: "${IMAGE_WIN_11}"
+  #         machineType: "${GCP_WIN_MACHINE_TYPE}"
+  #         disk_size: 100
+  #         disk_type: "pd-ssd"
+  #       artifact_paths:
+  #         - "$BEATS_PROJECT_NAME/build/*.xml"
+  #         - "$BEATS_PROJECT_NAME/build/*.json"
+  #       notify:
+  #         - github_commit_status:
+  #             context: "$BEATS_PROJECT_NAME: Windows 11 Unit Tests"
+
+  #     - label: ":windows: Windows 2019 Unit Tests"
+  #       command: |
+  #         Set-Location -Path $BEATS_PROJECT_NAME
+  #         mage build unitTest
+  #       key: "extended-win-2019-unit-tests"
+  #       agents:
+  #         provider: "gcp"
+  #         image: "${IMAGE_WIN_2019}"
+  #         machineType: "${GCP_WIN_MACHINE_TYPE}"
+  #         disk_size: 100
+  #         disk_type: "pd-ssd"
+  #       artifact_paths:
+  #         - "$BEATS_PROJECT_NAME/build/*.xml"
+  #         - "$BEATS_PROJECT_NAME/build/*.json"
+  #       notify:
+  #         - github_commit_status:
+  #             context: "$BEATS_PROJECT_NAME: Windows 2019 Unit Tests"
+  ####### skip: "see elastic/beats#23957 and elastic/beats#23958"
+
+  - group: "MacOs Extended Tests"
+    key: "extended-tests-macos"
+    if: build.env("BUILDKITE_PULL_REQUEST") == "false" || build.env("GITHUB_PR_LABELS") =~ /.*macOS.*/
+    steps:
+      - label: ":mac: MacOS Unit Tests"
+        key: "extended-macos-unit-tests"
+        command: ".buildkite/scripts/unit_tests.sh"
```

Comment:
could you check if you can switch to the pattern:

```
       command: |
          set -euo pipefail
          source .buildkite/scripts/install_macos_tools.sh
          cd metricbeat && mage build unitTest
```

?

see https://github.com/elastic/beats/pull/38827/files#diff-87cc9e7aecd3c1a6382e951a1b17429443aa3f12387328521fa06b2c8f83a814R220-R222

doing that will not require setting `$BEATS_PROJECT_NAME` in this PR (and let us simplify things in common scripts)

Reply:
turns out we don't need this for nodejs; only for linux unittests and go int tests: https://github.com/elastic/beats/blob/ci_xpack_heartbeat_monorepo/x-pack/heartbeat/Jenkinsfile.yml#L18